core/src/codechain_machine.rs

```diff
@@ -22,22 +22,23 @@ use cstate::{StateError, TopState, TopStateView};
 use ctypes::errors::{HistoryError, SyntaxError};
 use ctypes::machine::{Machine, WithBalances};
 use ctypes::transaction::{Action, AssetTransferInput, OrderOnTransfer, Timelock};
-use ctypes::BlockNumber;
+use ctypes::{BlockNumber, CommonParams};
+use parking_lot::{MappedRwLockReadGuard, RwLock, RwLockReadGuard};
 
 use crate::block::{ExecutedBlock, IsBlock};
 use crate::client::{BlockInfo, TransactionInfo};
 use crate::error::Error;
 use crate::header::Header;
-use crate::scheme::CommonParams;
 use crate::transaction::{SignedTransaction, UnverifiedTransaction};
 
 struct Params {
+    seq: usize,
```

Comment:
It seems that it is tied to the `seq` of the `metadata`. Should we change the name to `metadata_seq`?